src/docfx/lib/markdown/validation/DocsValidationExtension.cs

```diff
@@ -88,9 +53,90 @@ internal static class DocsValidationExtension
                 {
                     contentValidator.ValidateHeadings(inclusion, inclusionNodes, true);
                 }
+
+                foreach (var codeBlock in codeBlockItemList)
+                {
+                    contentValidator.ValidateCodeBlock(currentFile, codeBlock.codeBlockItem, codeBlock.IsIncluded);
+                }
             });
         }
 
+        private static void BuildHeadingNodes(
+            MarkdownObject node,
+            MarkdownVisitContext context,
+            MarkdownEngine markdownEngine,
+            List<ContentNode> documentNodes,
+            Dictionary<Document, List<ContentNode>> inclusionDocumentNodes,
+            bool? isCanonicalVersion)
+        {
+            ContentNode? documentNode = null;
+            if (node is HeadingBlock headingBlock)
+            {
+                var headingNode = CreateValidationNode<Heading>(context, isCanonicalVersion, headingBlock);
+
+                headingNode.Level = headingBlock.Level;
+                headingNode.Content = GetHeadingContent(headingBlock); // used for reporting
+                headingNode.HeadingChar = headingBlock.HeaderChar;
+                headingNode.RenderedPlainText = markdownEngine.ToPlainText(headingBlock); // used for validation
+                headingNode.IsVisible = MarkdigUtility.IsVisible(headingBlock);
+
+                documentNode = headingNode;
+            }
+            else if (node is LeafBlock leafBlock)
+            {
+                var contentNode = CreateValidationNode<ContentNode>(context, isCanonicalVersion, leafBlock);
+
+                contentNode.IsVisible = MarkdigUtility.IsVisible(leafBlock);
+
+                documentNode = contentNode;
+            }
+
+            if (documentNode != null)
+            {
+                documentNodes.Add(documentNode);
+                if (context.IsInclude)
+                {
+                    if (!inclusionDocumentNodes.TryGetValue(context.Document, out var inclusionNodes))
+                    {
+                        inclusionDocumentNodes[context.Document] = inclusionNodes = new List<ContentNode>();
+                    }
+
+                    inclusionNodes.Add(documentNode);
+                }
+            }
+        }
+
+        private static void BuildCodeBlockNodes(
+            MarkdownObject node,
+            MarkdownVisitContext context,
+            List<(bool IsInclude, CodeBlockItem codeBlockItem)> codeBlockItemList,
+            bool? isCanonicalVersion)
+        {
+            CodeBlockItem? codeBlockItem = null;
+
+            if (node is FencedCodeBlock || node.GetType().Name.Equals(nameof(CodeBlock)))
```

Comment:
consider change `GetType().Name.Equals` check into `is CodeBlock`, or better `switch node` for `FencedCodeBlock` and `CodeBlock`

Reply:
OK